src/blockindices.jl

```diff
@@ -198,20 +198,20 @@ represents a cartesian range inside a block.
 """
 BlockIndexRange
 
-BlockIndexRange(block::Block{N}, inds::Vararg{AbstractUnitRange{<:Integer},N}) where {N} =
+BlockIndexRange(block::Block{N}, inds::Vararg{AbstractVector,N}) where {N} =
```

Comment:
This generalization enables constructing non-contiguous subblock slices such as `Block(2)[[1, 3]]`. My biggest concern about this change is that `BlockIndexRange` is kind of a strange name for that object since it isn't a range. I can't really think of a better name though. One thing to consider could be renaming this struct to something more general like `BlockIndexVector` (or maybe `BlockIndices`, though that clashes with an alternative definition of `BlockIndices` proposed in #356) and then define `BlockIndexRange` as a type alias where the indices are constrained to `AbstractUnitRange{<:Integer}`.